src/services/codefixes/wrapJsxInFragment.ts

```diff
@@ -6,7 +6,7 @@ namespace ts.codefix {
         errorCodes,
         getCodeActions: context => {
             const { jsx } = context.program.getCompilerOptions();
-            if (jsx !== JsxEmit.React && jsx !== JsxEmit.ReactNative) {
+            if (!jsx || [JsxEmit.React, JsxEmit.ReactNative, JsxEmit.ReactJSX, JsxEmit.ReactJSXDev].indexOf(jsx) === -1) {

```

Comment:
Why isn’t this allowed in `JsxEmit.Preserve`? The error still applies and the fix is still plausibly valid. Unless someone knows a good reason not to offer the fix, I would just do

```suggestion
            if (!jsx) {
```

And, by the way, even that should be unnecessary, since you should never get this error if you have `undefined` or `JsxEmit.None` anyway 🤔